src/Integration.Vsix/CFamily/CFamilyHelper.cs

```diff
@@ -266,17 +266,12 @@ private static bool ExecuteAnalysis(IProcessRunner runner, string fileName, stri
             return success;
         }
 
-        internal /* for testing*/ static int GetTimeoutInMs()
+        internal /* for testing*/ static int GetTimeoutInMs(IEnvironmentSettings environmentSettings = null)
```

Comment:
The `CFamilyHelper` is static which made injecting the mock `IEnvironmentSetting` tricky.
In the end I went for making the env setting optional, with a fall-back to the static singleton `SharedEnvSettings`.

The test `SubProcessTimeout_NoSettingsSupplied_NoError` is just that the code doesn't fail when the optional argument isn't supplied.